tools/clang/lib/SPIRV/SpirvEmitter.cpp

```diff
@@ -12539,6 +12545,29 @@ SpirvEmitter::processSpvIntrinsicCallExpr(const CallExpr *expr) {
   return retVal;
 }
 
+SpirvInstruction *
+SpirvEmitter::processIntrinsicExecutionMode(const CallExpr *expr) {
+  llvm::SmallVector<uint32_t, 2> execModesParams;
+  uint32_t exeMode = 0;
+  const auto args = expr->getArgs();
+  for (uint32_t i = 0; i < expr->getNumArgs(); ++i) {
```

Comment:
It seems that I missed the extra operands of `OpExecutionMode` when I wrote the spec for `vk::ext_execution_mode(uint execution_mode);`. I am sorry about that.

This code looks good but I am concerned about
```
    SpirvConstantInteger *argInst =
        dyn_cast<SpirvConstantInteger>(doExpr(args[i]));
```

because they is no guarantee that `args[i]` is an expression to generate a constant integer. In addition, I am not sure we conduct the constant folding in DXC (definitely, spirv-opt does ..).

Why don't we just check whether `args[i]` is a constant integer or not here?
If it is not a constant integer, we can just report error and returns this method with `nullptr`. Please see `emitError(..)` in this file for the error report.